phoenix-shared/src/commonTest/kotlin/fr/acinq/phoenix/db/SqlitePaymentsDatabaseTest.kt

```diff
@@ -203,21 +203,27 @@ class SqlitePaymentsDatabaseTest {
         p.parts.forEach { assertEquals(onePartFailed, db.getOutgoingPart(it.id)) }
 
         // We should never update non-existing parts.
-        assertFails { db.updateOutgoingPart(UUID.randomUUID(), Either.Right(TemporaryNodeFailure)) }
-        assertFails { db.updateOutgoingPart(UUID.randomUUID(), randomBytes32()) }
+        assertFalse { db.outQueries.updateOutgoingPart(UUID.randomUUID(), Either.Right(TemporaryNodeFailure), 110) }
+        assertFalse { db.outQueries.updateOutgoingPart(UUID.randomUUID(), randomBytes32(), 110) }
 
         // Other payment parts are added.
         val newParts = listOf(
             OutgoingPayment.Part(UUID.randomUUID(), 5_000.msat, listOf(HopDesc(Lightning.randomKey().publicKey(), Lightning.randomKey().publicKey())), OutgoingPayment.Part.Status.Pending, 115),
             OutgoingPayment.Part(UUID.randomUUID(), 10_000.msat, listOf(HopDesc(Lightning.randomKey().publicKey(), Lightning.randomKey().publicKey())), OutgoingPayment.Part.Status.Pending, 120),
         )
         // Parts need a valid parent.
-        // NB: This test fails on iOS !
-        // The problem is that foreign key constraints are disabled.
-        // See iosDbFactory.kt for discussion.
-        assertFails { db.addOutgoingParts(UUID.randomUUID(), newParts) }
-        // New parts must have a unique id.
-        assertFails { db.addOutgoingParts(onePartFailed.id, newParts.map { it.copy(id = p.parts[0].id) }) }
+        if (isIOS()) {
+            // This is a known bug in SQLDelight on iOS.
+            // The problem is that foreign key constraints are disabled.
+            // See iosDbFactory.kt for discussion.
```

Comment:
The test runs fine on `iosX64` but not on `android` with the task `testDebugUnitTest` (on a Mac machine, but where `isIOS()` returns false) where it fails with the same `[SQLITE_CONSTRAINT_FOREIGNKEY]` exception. I'm not sure why I don't know if there's a good solution for this.

Reply:
Fixed in 3b7c28d